tests/components/spotify/test_search_string_generator.py

```diff
@@ -0,0 +1,152 @@
+    assert mapper.spotify_category_mapping is not None
+
+
+def test_init_invalid_file() -> None:
+    """Test initializing WeatherPlaylistMapper with an invalid mapping file."""
+    with pytest.raises(FileNotFoundError):
+        WeatherPlaylistMapper(
+            "homeassistant/components/spotify/test_spotify_mappings.json"
+        )
+
+
+def test_map_weather_to_playlists_valid_conditions(
+    mapper: WeatherPlaylistMapper,
+) -> None:
+    """Test mapping weather conditions to search string with valid conditions."""
+    search_string = mapper.map_weather_to_playlists(20, "pouring")
+    # Expected for 'warm' 'pouring'
+    assert search_string == "Rainy Day Rhythms"
+
+
+def test_map_weather_to_playlists_invalid_condition(
+    mapper: WeatherPlaylistMapper,
+) -> None:
+    """Test mapping weather conditions to search string with an invalid condition."""
+    with pytest.raises(ValueError):
+        mapper.map_weather_to_playlists(20, "sleepy")
+
+
+def test_map_weather_to_playlists_boundary_temperature_1(
+    mapper: WeatherPlaylistMapper,
+) -> None:
+    """Test mapping weather conditions to search string at the boundary temperature."""
+    search_string = mapper.map_weather_to_playlists(15, "cloudy")
+    # Expected for 'warm' 'cloudy
+    assert search_string == "Overcast Moods"
+
+
+def test_map_weather_to_playlists_boundary_temperature_2(
+    mapper: WeatherPlaylistMapper,
+) -> None:
+    """Test mapping at boundary temperature values."""
+    search_string = mapper.map_weather_to_playlists(0, "fog")
+    # Expected for 'cold' 'fog'
+    assert search_string == "Foggy Night Chill"
+
+
+def test_map_weather_to_playlists_negative_temperature(
+    mapper: WeatherPlaylistMapper,
+) -> None:
+    """Test mapping with negative temperature."""
+    search_string = mapper.map_weather_to_playlists(-5, "snowy")
+    # Expected for 'cold' 'snowy'
+    assert search_string == "Blizzard Ballads"
+
+
+def test_map_weather_to_playlists_high_temperature(
+    mapper: WeatherPlaylistMapper,
+) -> None:
+    """Test mapping with high temperature."""
+    search_string = mapper.map_weather_to_playlists(35, "fog")
+    # Expected for 'warm' 'fog'
+    assert search_string == "Misty Morning Mix"
+
+
+def test_map_weather_to_playlists_unusual_condition(
+    mapper: WeatherPlaylistMapper,
+) -> None:
+    """Test mapping with unusual weather condition."""
+    search_string = mapper.map_weather_to_playlists(20, "exceptional")
+    # Expected for 'warm' 'exceptional'
+    assert search_string == "Extraordinary Sounds"
+
+
+def test_map_weather_to_playlists_non_standard_condition_string(
+    mapper: WeatherPlaylistMapper,
+) -> None:
+    """Test mapping with non-standard condition strings."""
+    search_string = mapper.map_weather_to_playlists(20, "   SUNNY   ")
+    # Expected for 'warm' 'sunny'
+    assert search_string == "Sunny Day Play"
+
+
+def test_map_weather_to_playlists_large_temperature(
```

Comment:
In general: very nicely written and documented code! The only small thing I was a bit confused over was this testing function, you already have a function that is testing high temperature and for this one you are testing even higher temperature (100) which sounds a bit unreasonable 😅. Maybe this function should be responsible to check if the "warm lightning" condition works but with more reasonable high degrees, or did I missunderstand your code?🤔